securedrop/journalist.py

```diff
@@ -575,18 +575,38 @@ def download_single_submission(sid, fn):
 @app.route('/reply', methods=('POST',))
 @login_required
 def reply():
+    msg = request.form['msg']
+
+    # Reject empty replies
+    if not msg:
+        flash("You cannot send an empty reply!", "error")
+        return redirect(url_for('col', sid=g.sid))
+
     g.source.interaction_count += 1
     filename = "{0}-{1}-reply.gpg".format(g.source.interaction_count,
                                           g.source.journalist_filename)
-    crypto_util.encrypt(request.form['msg'],
+    crypto_util.encrypt(msg,
                         [crypto_util.getkey(g.sid), config.JOURNALIST_KEY],
                         output=store.path(g.sid, filename))
     reply = Reply(g.user, g.source, filename)
-    db_session.add(reply)
-    db_session.commit()
 
-    flash("Thanks! Your reply has been stored.", "notification")
-    return redirect(url_for('col', sid=g.sid))
+    try:
+        db_session.add(g.source)
```

Comment:
I think as written this will throw an `InvalidRequestError` because the underlying source has been deleted. But stepping back a bit: also recall that adding back a source doesn't regenerate the source directory / source GPG key that will be deleted when the source is deleted, so this line can be removed.